src/engine.rs

```diff
@@ -613,12 +620,19 @@ impl Engine {
             }
         };
 
-        let (values, logs, promises) = executor.into_state().deconstruct();
+        let (values, logs) = executor.into_state().deconstruct();
+        for log in logs {
```

Comment:
So this is the same logging mechanism that any smart contract on the EVM can use? This is not secure since it means any contract could create a log with the right topics to create a promise without having sent ETH to the precompile address.

Reply:
As discussed with @mfornet , we can securely determine which logs are created by our precompiles by using the `address` field instead of `topics`.